include/pokemon.h

```diff
@@ -350,6 +350,11 @@ struct BattleMove
     u8 zMoveEffect;
 };
 
+#define MOVE_EFFECT_WITH_DATA(moveEffect, chance) (moveEffect | (chance << 7)) // stores extra data in bits 8-14
+#define CURRENT_MOVE_EFFECT ((gBattleScripting.moveEffect << 2) >> 2)
+#define ARG_MOVE_EFFECT_ONLY(move) (gBattleMoves[move].argument & 0xC07F) // removes data in bits 8-14
+#define ARG_MOVE_EFFECT_DATA(moveEffect) ((moveEffect << 2) >> 9)
+
```

Comment:
Given that I can't make an anonymous union and a named one would require a _lot_ more refactoring (maybe another time), this cheat allows us to store data in the seven bits between the actual move effect (since none are higher than 127) and the MOVE_EFFECT_CERTAIN/MOVE_EFFECT_AFFECTS_USER bits.